File: apps/analysis-service/app/enums/emotion_enum.py
from enum import Enum

class EmotionEnum(str, Enum):
    JOY = 'joy'
    SADNESS = 'sadness'
    ANGER = 'anger'
    FEAR = 'fear'
    DISGUST = 'disgust'
    SURPRISE = 'surprise'
    NEUTRAL = 'neutral'

class IntensityLevelEnum(str, Enum):
    MILD = 'mild'
    MODERATE = 'moderate'
    SEVERE = 'severe'

class DominantModalityEnum(str, Enum):
    TEXT = 'text'
    IMAGE = 'image'
    # VIDEO = 'video'

class EmotionTimeWindowEnum(str, Enum):
    LAST_24_HOURS = '24h'  # e.g., last 24 hours
    LAST_7_DAYS = '7d'  # e.g., last 7 days
    LAST_30_DAYS = '30d'  # e.g., last 30 days

class RiskHintLevelEnum(str, Enum):
    NONE = 'none'
    WEAK = 'weak'
    MEDIUM = 'medium'
    HIGH = 'high'

File: packages/dtos/src/emotion/enums/emotion-enum.ts
export enum IntensityLevel {
  MILD = 'mild',
  MODERATE = 'moderate',
  SEVERE = 'severe',
}

export enum DominantModality {
  TEXT = 'text',
  IMAGE = 'image',
  //VIDEO = 'video',
}

export enum EmotionTimeWindow {
  ONE_DAY = '1d',
  SEVEN_DAYS = '7d',
  THIRTY_DAYS = '30d',
}

export enum LowCaseEmotion {
  JOY = 'joy',
  SADNESS = 'sadness',
  ANGER = 'anger',
  FEAR = 'fear',
  DISGUST = 'disgust',
  SURPRISE = 'surprise',
  NEUTRAL = 'neutral',
}

export enum RiskLevel {
  NORMAL = 'NORMAL',
  MILD_STRESS = 'MILD_STRESS',
  MODERATE_RISK = 'MODERATE_RISK',
  HIGH_RISK = 'HIGH_RISK',
  CRISIS = 'CRISIS',
}

export enum TriggerFlag {
  SUICIDAL_IDEATION = 'SUICIDAL_IDEATION',
  LONG_TERM_SADNESS = 'LONG_TERM_SADNESS',
  HIGH_ANXIETY_BURST = 'HIGH_ANXIETY_BURST',
  SARCASM_CONFLICT = 'SARCASM_CONFLICT',
}

export enum RiskHintLevel {
  NONE = 'none',
  LOW = 'low',
  MEDIUM = 'medium',
  HIGH = 'high',
}

export enum InsightType {
  ABOVE_BASELINE = 'ABOVE_BASELINE',
  DETERIORATING_TREND = 'DETERIORATING_TREND',
  HIGH_NEGATIVITY = 'HIGH_NEGATIVITY',
  HIGH_RISK = 'HIGH_RISK',
  HIGH_VOLATILITY = 'HIGH_VOLATILITY',
  NEGATIVE_STREAK = 'NEGATIVE_STREAK',
  NORMALIZING = 'NORMALIZING',
  POSITIVE_STATE = 'POSITIVE_STATE',
  RECOVERING_TREND = 'RECOVERING_TREND',
  STABLE_STATE = 'STABLE_STATE',
}

export enum InsightTone {
  POSITIVE = 'positive',
  NEUTRAL = 'neutral',
  WARNING = 'warning',
  CRITICAL = 'critical',
}
